virl2_client/models/auth_management.py

```diff
@@ -21,34 +21,37 @@
 from __future__ import annotations
 
 import time
-from enum import Enum
-from typing import Any, Optional
-
-from httpx import Client
+from typing import Any, Optional, TYPE_CHECKING
 
 from .resource_pools import ResourcePool
+from ..exceptions import MethodNotActive
 
-
-class AuthMethod(Enum):
-    LOCAL = "local"
-    LDAP = "ldap"
+if TYPE_CHECKING:
+    from httpx import Client
 
 
 _BASE_URL = "system/auth"
+_CONFIG_URL = _BASE_URL + "/config"
+_TEST_URL = _BASE_URL + "/test"
 
 
 class AuthManagement:
-    _METHOD_URL = _BASE_URL + "/config"
-
     def __init__(self, session: Client, auto_sync=True, auto_sync_interval=1.0):
+        """
+        Sync and modify authentication settings.
+
+        :param session: parent client's httpx.Client object
+        :param auto_sync: whether to automatically synchronize resource pools
+        :param auto_sync_interval: how often to synchronize resource pools in seconds
+        """
         self.auto_sync = auto_sync
         self.auto_sync_interval = auto_sync_interval
         self._last_sync_time = 0.0
         self._session = session
-        self._method = AuthMethod.LOCAL
-        self._managers: dict[AuthMethod, Optional[AuthMethodManager]] = {
-            AuthMethod("local"): None,
-            AuthMethod("ldap"): LDAPManager(self),
+        self._settings = {}
```

Comment:
No settings are set by default, we need to call sync or set {"method": "local"} here.  Otherwise requesting an attribute will fail with KeyError.

Or we can do this:
~~~
- self._settings[setting]
+ self._settings.get(setting)
~~~

Reply:
There are a few occurrences of `self._settings["method"]` too.

Reply:
I don't see anywhere where _settings are accessed without calling sync_if_outdated first, do you?

Reply:
Right, the issue was when sync is off... Submitted a fix which syncs if settings are empty even with sync off, we need the initial data either way